### PR TITLE
Fix all left mgmt spotbugs

### PR DESCRIPTION
**Pull request by ChenTanyi**
17 NM_CONFUSING left:
* com.azure.management.resources.GenericResourceFilter.tagname() and com.azure.management.resources.ResourceGroupFilter.tagName()
* com.azure.management.resources.GenericResourceFilter.tagvalue() and com.azure.management.resources.ResourceGroupFilter.tagValue()
* com.azure.management.storage.AzureEntityResource.etag() and com.azure.management.storage.BlobContainersCreateOrUpdateImmutabilityPolicyHeaders.eTag()
* com.azure.management.network.ApplicationGatewayFrontendIPConfiguration.privateIPAddress() and com.azure.management.network.VpnClientConnectionHealthDetail.privateIpAddress()
* com.azure.management.network.HubIPAddresses.publicIPAddresses() and com.azure.management.network.models.NatGatewayInner.publicIpAddresses()
* com.azure.management.network.implementation.ExpressRouteCrossConnectionImpl.sTag() and com.azure.management.network.models.ExpressRouteCircuitInner.stag()
* com.azure.management.sql.implementation.SqlSyncMemberImpl.userName() and com.azure.management.sql.models.JobCredentialInner.username()
* com.azure.management.appservice.DetectorAbnormalTimePeriod.metaData() and com.azure.management.appservice.Solution.metadata()
* com.azure.management.appservice.DiagnosticMetricSample.timestamp() and com.azure.management.appservice.models.CertificateEmailInner.timeStamp() 
* com.azure.management.appservice.StaticSitePatchResource.defaultHostname() and com.azure.management.appservice.implementation.WebAppBaseImpl.defaultHostName()
* com.azure.management.appservice.models.SiteConfigInner.publishingUsername() and com.azure.management.appservice.models.UserInner.publishingUserName()
* com.azure.management.containerservice.ManagedClusterAADProfile.tenantID() and com.azure.management.containerservice.ManagedClusterIdentity.tenantId()
* com.azure.management.monitor.ActivityLogAlertActionGroup.webhookProperties() and com.azure.management.monitor.MetricAlertAction.webHookProperties()
* com.azure.management.monitor.MetricAlertStatusProperties.timestamp() and com.azure.management.monitor.MetricValue.timeStamp()
* com.azure.management.monitor.ScaleAction.cooldown() and com.azure.management.monitor.implementation.ScaleRuleImpl.coolDown()
* com.azure.management.monitor.implementation.MetricCollectionImpl.resourceRegion() and com.azure.management.monitor.models.ResponseInner.resourceregion()
* com.azure.management.dns.implementation.DnsRecordSetImpl.eTag() and com.azure.management.dns.models.RecordSetInner.etag()

### CODE REVIEW

**Comment by ChenTanyi** [comment]
* tagname
* tagvalue
* etag
* privateIPAddress
* publicIPAddresses
* sTag
* userName
* metaData
* timestamp
* defaultHostname
* publishingUsername
* tenantID
* webhookProperties
* cooldown
* resourceRegion

File: sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/GalleryImageVersionImpl.java [hunk 1]
```diff
@@ -360,7 +361,7 @@ private static String getValueFromIdByName(String id, String name) {
         Iterator<String> itr = iterable.iterator();
         while (itr.hasNext()) {
             String part = itr.next();
-            if (part != null && part.trim() != "") {
+            if (part != null && !part.trim().equals("")) {
```
Comment (weidongxu-microsoft) [inline sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/GalleryImageVersionImpl.java:364]: `!part.trim().isEmpty()` ?

File: sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/VirtualMachineScaleSetImpl.java [hunk 2]
```diff
@@ -1036,7 +1034,7 @@ public VirtualMachineScaleSetImpl withoutOverProvisioning() {
 
     @Override
     public VirtualMachineScaleSetImpl withCapacity(int capacity) {
```
Comment (weidongxu-microsoft) [inline sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/VirtualMachineScaleSetImpl.java:1036]: Should it take long?

File: sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/GalleryImageVersionImpl.java (hunk 1 above)
Comment (ChenTanyi) [reply sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/GalleryImageVersionImpl.java]: done

File: sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/VirtualMachineScaleSetImpl.java (hunk 2 above)
Comment (ChenTanyi) [reply sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/VirtualMachineScaleSetImpl.java]: done

**Pull request merged**